frontend/src/metabase/parameters/components/ParameterLinkedFilters/ParameterLinkedFilters.tsx

```diff
@@ -111,6 +101,42 @@ const ParameterLinkedFilters = ({
   );
 };
 
+function getDisableReason({
```

Comment:
This function is actually a component.
It can't be used as a component (i.e. `<DisableReason />`) because we're checking whether it returns `null`.
But there's another rendering path when it returns `null`.

I'd refactor it a bit, to keep it in React spirit.

1. Extract code from the first `if` to a separate component, e.g. `NoUsableParameters`.
2. Extract code from the second `if` to a separate component, e.g. `ParametersFromOtherSource`
3. Extract code after `disableReason ||` to a separate component, e.g. `UsableParameters` (and move hooks there, as that's where they're needed)
4. Refactor this function to a component, like this:

```tsx
const Content = (/* props omitted for brevity */) => {
  if (usableParameters.length === 0) {
    return <NoUsableParameters onShowAddParameterPopover={onShowAddParameterPopover} />;
  }

  if (parameter.values_source_type != null) {
    return <ParametersFromOtherSource />;
  }

  return (
    <UsableParameters
      parameter={parameter}
      usableParameters={usableParameters}
      onChangeFilteringParameters={onChangeFilteringParameters}
      onShowAddParameterPopover={onShowAddParameterPopover}
    />
  );
};
```
5. Use this component unconditionally in `ParameterLinkedFilters`

WDYT @calherries ?

Reply:
Makes a lot of sense. I'll look out for this kind of refactoring in the future, thanks